### PR TITLE
Apple Pay Fixes

### PR DESCRIPTION
**Pull request by DerAusHeidelberg**
- JSON can now be converted into Object (Added Empty Constructors for Apple Pay and the Apple Pay Header)

### CODE REVIEW

**Review by Ryouzanpaku — APPROVED** [review]
LGTM.

File: src/test/java/com/unzer/payment/business/paymenttypes/ApplepayTest.java [hunk 1]
```diff
@@ -40,6 +42,17 @@
 
 public class ApplepayTest extends AbstractPaymentTest {
 
+    @Test
+    public void testCreateApplePayTypeFromJson() throws JsonProcessingException {
```
Comment (sixer1182) [inline src/test/java/com/unzer/payment/business/paymenttypes/ApplepayTest.java:46]: Is this the certificate?

File: src/main/java/com/unzer/payment/paymenttypes/Applepay.java [hunk 2]
```diff
@@ -49,11 +49,14 @@
     private String currencyCode;
     private BigDecimal transactionAmount;
 
-    public Applepay(String version, String data, String signature, ApplepayHeader applepayHeader) {
+    public Applepay() {
+    }
```
Comment (sixer1182) [inline src/main/java/com/unzer/payment/paymenttypes/Applepay.java:53]: What for?

File: src/main/java/com/unzer/payment/paymenttypes/ApplepayHeader.java [hunk 3]
```diff
@@ -25,6 +25,9 @@
     private String publicKeyHash;
     private String transactionId;
 
+    public ApplepayHeader() {
+    }
```
Comment (sixer1182) [inline src/main/java/com/unzer/payment/paymenttypes/ApplepayHeader.java:29]: What for?

File: src/test/java/com/unzer/payment/business/paymenttypes/ApplepayTest.java (hunk 1 above)
Comment (DerAusHeidelberg) [reply src/test/java/com/unzer/payment/business/paymenttypes/ApplepayTest.java:46]: This is the Response JSON from Apple after the Payment Authorization. But the Values have been masked on different parts.

File: src/main/java/com/unzer/payment/paymenttypes/Applepay.java (hunk 2 above)
Comment (DerAusHeidelberg) [reply src/main/java/com/unzer/payment/paymenttypes/Applepay.java:53]: This is for the JSON-Library which we are using. This needs an empty public constructor to be able to map a JSON into this object.

File: src/main/java/com/unzer/payment/paymenttypes/ApplepayHeader.java (hunk 3 above)
Comment (DerAusHeidelberg) [reply src/main/java/com/unzer/payment/paymenttypes/ApplepayHeader.java:29]: Same as for the ApplePay-Constructor

File: src/main/java/com/unzer/payment/paymenttypes/Applepay.java (hunk 2 above)
Comment (sixer1182) [reply src/main/java/com/unzer/payment/paymenttypes/Applepay.java:53]: ok

**Pull request merged**